sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/configuration/SpringDocConfiguration.java

```diff
@@ -0,0 +1,144 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package fr.insee.sugoi.services.configuration;
+
+import io.swagger.v3.oas.models.Components;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.info.Contact;
+import io.swagger.v3.oas.models.info.Info;
+import io.swagger.v3.oas.models.info.License;
+import io.swagger.v3.oas.models.security.OAuthFlow;
+import io.swagger.v3.oas.models.security.OAuthFlows;
+import io.swagger.v3.oas.models.security.SecurityRequirement;
+import io.swagger.v3.oas.models.security.SecurityScheme;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springdoc.core.customizers.OperationCustomizer;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class SpringDocConfiguration {
+  private static final Logger logger = LoggerFactory.getLogger(SpringDocConfiguration.class);
+
+  @Value("${fr.insee.sugoi.springdoc.issuer.url:}")
+  public String issuerURL;
+
+  public final String OAUTHSCHEME = "oAuthScheme";
+  public final String SCHEMEBASIC = "basic";
+
+  @Bean
+  @ConditionalOnProperty(
+      name = "fr.insee.sugoi.security.bearer-authentication-enabled",
+      havingValue = "true",
+      matchIfMissing = false)
+  public OpenAPI customOpenAPIOIDC() {
+    final OpenAPI openapi = createOpenAPI();
+    openapi.components(
+        new Components()
+            .addSecuritySchemes(
+                OAUTHSCHEME,
+                new SecurityScheme()
+                    .type(SecurityScheme.Type.OAUTH2)
+                    .in(SecurityScheme.In.HEADER)
+                    .description("Authentification keycloak")
+                    .flows(
+                        new OAuthFlows()
+                            .authorizationCode(
+                                new OAuthFlow()
+                                    .authorizationUrl(issuerURL + "/protocol/openid-connect/auth")
```

Comment:
this should be customizable in case we use another implementation than keycloak.